### ISSUE TITLE
No error when api doesn't exist in boto3

### ISSUE THREAD

**Issue by pju-d2si** [cloud-custodian/cloud-custodian#4412]
Hi,
for help developement of cloudcustodian, i return you my last experience.

i found a little problem with policy custodian and function not present on a old boto3 version.

in my case, i check tag-user with my filter with orginal version on lambda (old version),tag users it's a "recent features" on boto3.

when i execute my filter custodian, it return 0 users with my tag-custom tag (indeed my lib boto is not update) but cloudcustodian don't return an error if function exist in custodian and not in boto libs.

when i have added a layer boto-latest-version, my filter working fine

i think doen't have a suffisente skill for commit a good change for this problem.

thank for your support
best regards

**Comment by kapilt** [comment]
I think our best path forward is creating layers on demand with latest botocore layers (or at least the executing cli's latest).

**Comment by pju-d2si** [comment]
yes i have set this solution with layer for boto, but I wanted to go back my experience when a lib is not updated, i haved not error message if api doesn't exist.

thanks you for your return and help

**Comment by kapilt** [comment]
there's some historical reasons why we don't do this when fetching resources vs returning an empty set. its worth revisiting potentially, but realistically ensuring we always have the right library version would also resolve. errors around filters, actions would propagate.
